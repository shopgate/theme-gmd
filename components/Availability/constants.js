/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * The availability states of a product.
 */
export const AVAILABILITY_STATE_OK = 'ok';
export const AVAILABILITY_STATE_WARNING = 'warning';
export const AVAILABILITY_STATE_ALERT = 'alert';
