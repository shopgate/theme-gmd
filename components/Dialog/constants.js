/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const DIALOG_TEXT_MESSAGE = 'DIALOG_TEXT_MESSAGE';
export const MODAL_VARIANT_SELECT = 'MODAL_VARIANT_SELECT';
