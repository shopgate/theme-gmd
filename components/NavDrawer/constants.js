/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const QUICKLINKS_MENU = 'quicklinks';
