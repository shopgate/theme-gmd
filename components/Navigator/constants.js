/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

 /**
  * Action Types.
  */

export const TOGGLE_NAVIGATOR_CART_ICON = 'TOGGLE_NAVIGATOR_CART_ICON';
export const TOGGLE_NAVIGATOR_SEARCH = 'TOGGLE_NAVIGATOR_SEARCH';
export const TOGGLE_NAV_DRAWER = 'TOGGLE_NAV_DRAWER';
export const TOGGLE_PROGRESS_BAR = 'TOGGLE_PROGRESS_BAR';
export const TOGGLE_LOGIN = 'TOGGLE_LOGIN';
export const SET_NAVIGATOR_BACKGROUND = 'SET_NAVIGATOR_BACKGROUND';
export const SET_NAVIGATOR_COLOR = 'SET_NAVIGATOR_COLOR';
export const SET_NAVIGATOR_SEARCH_QUERY = 'SET_NAVIGATOR_SEARCH_QUERY';
export const SET_FILTER_OPENED = 'SET_FILTER_OPENED';
export const SET_FILTER_CLOSED = 'SET_FILTER_CLOSED';
export const SET_FILTER_ATTRIBUTE_OPENED = 'SET_FILTER_ATTRIBUTE_OPENED';
export const SET_FILTER_ATTRIBUTE_CLOSED = 'SET_FILTER_ATTRIBUTE_CLOSED';
export const SET_SEARCH_ENABLED = 'SET_SEARCH_ENABLED';
export const SET_SEARCH_DISABLED = 'SET_SEARCH_DISABLED';
export const SET_ICON_SHADOW_ENABLED = 'SET_ICON_SHADOW_ENABLED';
export const SET_ICON_SHADOW_DISABLED = 'SET_ICON_SHADOW_DISABLED';
export const SET_TITLE_ENABLED = 'SET_TITLE_ENABLED';
export const SET_TITLE_DISABLED = 'SET_TITLE_DISABLED';
export const SET_NAVIGATOR_DISABLED = 'SET_NAVIGATOR_DISABLED';
export const SET_NAVIGATOR_ENABLED = 'SET_NAVIGATOR_ENABLED';
