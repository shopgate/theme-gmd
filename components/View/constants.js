export const SET_VIEW_TITLE = 'SET_VIEW_TITLE';
