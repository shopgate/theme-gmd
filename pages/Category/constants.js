/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const GRID_VIEW = 'grid';
export const LIST_VIEW = 'list';

export const SET_CATEGORY_VIEW_MODE = 'SET_CATEGORY_VIEW_MODE';
