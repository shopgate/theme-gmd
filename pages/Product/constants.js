/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Action Types
 */
export const OPEN_PRODUCT_GALLERY = 'OPEN_PRODUCT_GALLERY';
export const REQUEST_PRODUCT_DATA = 'REQUEST_PRODUCT_DATA';

/**
 * Max number of reviews shown
 * @type {number}
 */
export const REVIEW_PREVIEW_COUNT = 2;
