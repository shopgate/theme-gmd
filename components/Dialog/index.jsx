/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Modal from '@shopgate/pwa-common/components/Modal';
import Backdrop from '@shopgate/pwa-common/components/Backdrop';
import { MODAL_PIPELINE_ERROR } from '@shopgate/pwa-common/constants/ModalTypes';
import {
  DIALOG_TEXT_MESSAGE,
  MODAL_VARIANT_SELECT,
} from './constants';
import PipelineErrorDialog from './components/PipelineErrorDialog';
import TextMessageDialog from './components/TextMessageDialog';
import BasicDialog from './components/BasicDialog';
import VariantSelectModal from './components/VariantSelectModal';

/**
 * The main component for rendering dialogs.
 * This component takes care of choosing the correct component body for the given type
 * and render it on a modal overlay.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const Dialog = ({ modal, onConfirm, onDismiss }) => {
  // Assemble the actions.
  const actions = [];
  const { confirm, dismiss, title, params, message, type } = modal;

  if (confirm) {
    // We have a confirm label, add a confirm action.
    actions.push({
      label: modal.confirm,
      action: onConfirm,
    });
  }

  if (dismiss) {
    // We have a dismiss label, add a dismiss action.
    actions.push({
      label: modal.dismiss,
      action: onDismiss,
    });
  }

  let dialogType = type;

  if (!dialogType && message) {
    dialogType = DIALOG_TEXT_MESSAGE;
  }

  switch (dialogType) {
    case MODAL_PIPELINE_ERROR:
      return (
        <Modal>
          <Backdrop isVisible level={0} />
          <PipelineErrorDialog actions={actions} title={title} params={params} message={message} />
        </Modal>
      );
    case DIALOG_TEXT_MESSAGE:
      return (
        <Modal>
          <Backdrop isVisible level={0} />
          <TextMessageDialog actions={actions} title={title} params={params} message={message} />
        </Modal>
      );
    case MODAL_VARIANT_SELECT:
      return (
        <Modal>
          <Backdrop isVisible level={0} />
          <VariantSelectModal actions={actions} title={title} params={params} message={message} />
        </Modal>
      );
    default:
      return (
        <Modal>
          <Backdrop isVisible level={0} />
          <BasicDialog actions={actions} title={title} params={params} />
        </Modal>
      );
  }
};

Dialog.propTypes = {
  modal: PropTypes.shape({
    title: PropTypes.string,
    confirm: PropTypes.string,
    dismiss: PropTypes.string,
    message: PropTypes.string,
    params: PropTypes.shape(),
  }).isRequired,
  onConfirm: PropTypes.func,
  onDismiss: PropTypes.func,
};

Dialog.defaultProps = {
  onConfirm: () => {},
  onDismiss: () => {},
};

export default Dialog;
