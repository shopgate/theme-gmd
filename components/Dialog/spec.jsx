/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { MODAL_PIPELINE_ERROR } from '@shopgate/pwa-common/constants/ModalTypes';
import { MODAL_VARIANT_SELECT } from './constants';
import Dialog from './index';

describe('<Dialog />', () => {
  it('should render without props', () => {
    const wrapper = shallow(<Dialog modal={{ message: 'msg' }} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('TextMessageDialog').length).toBe(1);
  });

  it('should render BasicDialog when no message given', () => {
    const wrapper = shallow(<Dialog modal={{ message: null }} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('BasicDialog').length).toBe(1);
  });

  it('should render a special dialog', () => {
    const params = {
      errorCode: '',
      message: '',
      pipelineName: '',
      request: {},
    };

    const wrapper = shallow(<Dialog
      modal={{
        type: MODAL_PIPELINE_ERROR,
        params,
      }}
    />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('DefaultDialog').length).toBe(0);
    expect(wrapper.find('PipelineErrorDialog').length).toBe(1);
  });

  it('should render variant select dialog', () => {
    const params = {
      productId: 'product_1',
    };

    const wrapper = shallow(
      <Dialog
        modal={{
          message: 'Test',
          type: MODAL_VARIANT_SELECT,
          params,
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('DefaultDialog').length).toBe(0);
    expect(wrapper.find('VariantSelectModal').length).toBe(1);
  });
});
