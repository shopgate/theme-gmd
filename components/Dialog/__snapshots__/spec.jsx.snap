// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> should render BasicDialog when no message given 1`] = `
<Modal>
  <Backdrop
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <BasicDialog
    actions={Array []}
    title={null}
  />
</Modal>
`;

exports[`<Dialog /> should render a special dialog 1`] = `
<Modal>
  <Backdrop
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <PipelineErrorDialog
    actions={Array []}
    message=""
    params={
      Object {
        "errorCode": "",
        "message": "",
        "pipelineName": "",
        "request": Object {},
      }
    }
  />
</Modal>
`;

exports[`<Dialog /> should render variant select dialog 1`] = `
<Modal>
  <Backdrop
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <VariantSelectModal
    actions={Array []}
    message="Test"
    params={
      Object {
        "productId": "product_1",
      }
    }
    title={null}
  />
</Modal>
`;

exports[`<Dialog /> should render without props 1`] = `
<Modal>
  <Backdrop
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <TextMessageDialog
    actions={Array []}
    message="msg"
    title={null}
  />
</Modal>
`;
